plugins/linkattacher/src/linkattacherserverimpl.cpp

```diff
@@ -16,147 +16,155 @@ LinkAttacherServerImpl::LinkAttacherServerImpl()
-    gazebo::physics::Link_V robot_model_links = robot_model->GetLinks();
-    for(int i=0; i < robot_model_links.size(); i++)
+    gazebo::physics::Link_V child_model_links = child_model->GetLinks();
+    gazebo::physics::LinkPtr child_model_link;
+    for(int i=0; i < child_model_links.size(); i++)
     {
-        //E.g iCub::l_hand::l_hand_base_link
-
-        std::string candidate_robot_link_name = robot_model_links[i]->GetScopedName();
+        std::string candidate_child_model_link_name = child_model_links[i]->GetScopedName();
 
-        std::size_t lastcolon = candidate_robot_link_name.rfind(":");
-        std::string unscoped_robot_link_name = candidate_robot_link_name.substr(lastcolon+1,std::string::npos);
-
-        if(unscoped_robot_link_name == robot_link_name)
+        if(GazeboYarpPlugins::hasEnding(candidate_child_model_link_name, child_model_link_name))
         {
-            yInfo() << LogPrefix << "Attach: Full scoped Candidate robot link name: " << candidate_robot_link_name << " found";
-            gazebo::physics::LinkPtr robot_link = robot_model_links[i];
-            if(!robot_link)
-            {
-                yError() << LogPrefix << "Attach: " << robot_link_name << " is not found";
-                return false;
-            }
-            else
-            {
-              yInfo() << LogPrefix << "Attach: " << robot_link->GetName() << " found";
-            }
-
-            //This is joint creation
-            gazebo::physics::JointPtr joint;
+            child_model_link = child_model_links[i];
+            yInfo() << LogPrefix << "Attach: " << child_model_link->GetName() << " link found";
+            break;
+        }
+    }
+    if(!child_model_link)
+    {
+    	yError() << LogPrefix << "Attach: " << child_model_link_name << " link is not found";
+    	return false;
+    }
 
-            #if GAZEBO_MAJOR_VERSION >= 8
-            joint = _world->Physics()->CreateJoint("fixed",object_model);
-            #else
-            joint = _world->GetPhysicsEngine()->CreateJoint("fixed",object_model);
-            #endif
+    //This is joint creation
+    gazebo::physics::JointPtr joint;
 
-            if(!joint)
-            {
-                yError() << LogPrefix << "Attach: Unable to create joint";
-                return false;
-            }
+    #if GAZEBO_MAJOR_VERSION >= 8
+    joint = _world->Physics()->CreateJoint("fixed",parent_model);
+    #else
+    joint = _world->GetPhysicsEngine()->CreateJoint("fixed",parent_model);
+    #endif
 
-            std::string joint_name = model_link_name + "_magnet_joint";
-            joint->SetName(joint_name);
-            yInfo() << LogPrefix << "Attach: Magnet joint: " << joint->GetName() << " created";
+    if(!joint)
+    {
+        yError() << LogPrefix << "Attach: Unable to create joint";
+        return false;
+    }
 
-            joint->SetModel(object_model);
+    std::string joint_name = parent_model_link_name + "_magnet_joint";
```

Comment:
The `joint_name` variable name should be consistent with the joint name change above